test/timecop_test.rb

```diff
@@ -689,8 +689,105 @@ def test_thread_safe_timecop_returns_after_block
     Timecop.thread_safe = false
   end
 
+  if RUBY_VERSION >= '2.1.0'
+    def test_process_clock_gettime_monotonic
+      Timecop.freeze do
```

Comment:
@alexcwatt just playing around with what you've done. I'm not familiar with Process.clock_gettime but in this test when i add 

```
      refute_same monotonic, monotonic
```
Outside the freeze block the test fails.  If the assert_same line inside the block is to have meaning I think we need to be sure that it can't false positive.

When I change the test like this

```
      t1 = monotonic
      t2 = monotonic
      refute_same t1, t2
      Timecop.freeze do
        t1 = monotonic
        t2 = monotonic
        assert_same t1, t2, "CLOCK_MONOTONIC is not frozen"
      end
```
The result is what you're expecting. I think this is an issue with this test, no? And if that's the case then I maybe some of these other tests need to be looked at carefully. Please have a look.

Reply:
Actually that change doesn't matter. If i run it a few times it still fails occasionally. I guess we just need a sleep in between calls to monotonic.

Reply:
```
    def test_process_clock_gettime_monotonic
      t1 = monotonic
      sleep(0.1)
      t2 = monotonic
      refute_same t1, t2
      Timecop.freeze do
        t1 = monotonic
        sleep(0.1)
        t2 = monotonic
        assert_same t1, t2, "CLOCK_MONOTONIC is not frozen"
      end
```

Reply:
As mentioned at https://github.com/travisjeffery/timecop/pull/419#discussion_r1595497983, I would suggest setting an epsilon constant. We could also add a helper to get two monotonic:

```rb
class TestTimecop < Minitest::Test
  TIME_EPSILON = 0.001 # seconds

  def test_process_clock_gettime_monotonic
    t1, t2 = consecutive_monotonics
    refute_same t1, t2, "CLOCK_MONOTONIC should not be frozen"

    Timecop.freeze do
      t1, t2 = consecutive_monotonics
      assert_same t1, t2, "CLOCK_MONOTONIC should be frozen"
    end
  end

  private
  
  def monotonic
    Process.clock_gettime(Process::CLOCK_MONOTONIC)
  end

  def consecutive_monotonics
    t1 = monotonic
    sleep(TIME_EPSILON)
    t2 = monotonic

    [t1, t2]
  end
end
```

Reply:
Thanks for this feedback! I like the suggestion for a fix @lavoiesl.